sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
Another point: this doesn't stop someone from passing around `ResponseWithError` everywhere getting right back to the pattern you're trying to prevent.